parquet/parquet.cpp

```diff
@@ -1523,7 +1536,7 @@ static void dissect_map(pq_writer *pw, lua_State *lua, pq_node *n, int16_t r, in
   bool found = false;
   lua_pushnil(lua);
   while (lua_next(lua, -2) != 0) {
-    dissect_field(pw, lua, vn, cr, vn->dl);
```

Comment:
the following will resolve the issue, dissect just needs the proper parent node depth since we don't have a guaranteed leaf node anymore. : 
`dissect_field(pw, lua, vn, cr, n->dl);`